src/function/scalar/string/levenshtein.cpp

```diff
@@ -0,0 +1,84 @@
+#include "duckdb/function/scalar/string_functions.hpp"
+#include "duckdb/common/vector_operations/vector_operations.hpp"
+#include "duckdb/common/string_util.hpp"
+
+#include <ctype.h>
+#include <algorithm>
+
+namespace duckdb {
+
+// See: https://www.kdnuggets.com/2020/10/optimizing-levenshtein-distance-measuring-text-similarity.html
+// And: Iterative 2-row algorithm: https://en.wikipedia.org/wiki/Levenshtein_distance
+// Note: A first implementation using the array algorithm version resulted in an error raised by duckdb (too muach memory usage) 
+static int levenshtein2(const string& txt, const string& tgt) 
+    {
+    if (txt.size() < 1 || tgt.size() < 1) 
+    {
+        return -1; // too short
+    }
+	if (txt.size() > 254 || tgt.size() > 254) 
+    {
+        return -2; // too long
+    }
+
+    std::vector<int> distances(txt.size() + 1, 0);
+
+    for (int pos = 1; pos <= (int)txt.size(); pos++) 
```

Comment:
Could you change the indexes used in the different loops to `idx_t` rather than `int`? We prefer to avoid `int` since its size is platform-dependent.

Reply:
Certainly.